src/cpp/lpnamer/input_reader/WeightsFileWriter.cpp

```diff
@@ -54,7 +57,10 @@ void YearlyWeightsWriter::DumpMpsWeightsToFile() const {
   mps_weights_file.open(file);
 
   for (const auto& [mps_name, weight] : mps_weights_) {
-    mps_weights_file << mps_name.string() << " " << weight << std::endl;
+    mps_weights_file << SolverConfig::FileName(
+                            mps_name, SolverConfig(std::move(solver_name_)))
```

Comment:
Careful with `std::move`, it invalidates it's argument
```suggestion
                            mps_name, SolverConfig(solver_name_))
```

Reply:
Method `SolverConfig::FileName` can surely be improved

```cpp
// Current : use another instance
static std::filesystem::path SolverConfig::FileName(const SolverConfig& c);
// Proposed : use *this
std::filesystem::path SolverConfig::FileName() const;
```

Reply:
Suggestion applied